src/app/modules/react-migration/react-migration.controller.ts

```diff
@@ -15,6 +15,7 @@ export type SetEnvironmentParams = {
 
 export const RESPONDENT_COOKIE_OPTIONS = {
   httpOnly: false,
+  maxAge: 31 * 2 * 24 * 60 * 60, // 2 months
```

Comment:
was this what we decided? @timotheeg thoughts? or we can keep this first.

Reply:
sorry if the issue is unclear. basically the idea is this:
- if you are randomly assigned React, you should stay on React unless you opt for Angular (hence 2 months)
- if you are randomly assigned AngularJS, you should stay on AngularJS for the next 24h (so the rollout % can be increased the next day)
- if you opt for Angular, you should stay on Angular until the relevant rollout % is 100% (e.g. if I opt for Angular on an email mode form, I will stay on Angular for email mode forms until email mode form rollout is 100%, after which I can be on React. if I opt for Angular on storage mode form, I stay on Angular until storage mode is 100%)

is that clearer?

Reply:
That doesn't sound right 🤔 

> if you are randomly assigned React, you should stay on React unless you opt for Angular (hence 2 months)

This will cause the percentage to increase in favour of react more than what the rollout threshold suggests.

I think the random assignment should remain as session as we had discussed, and only explicit switches should have an expended expiry.

Reply:
Also the change here needs more work to fit the logic. `RESPONDENT_COOKIE_OPTIONS` is currently applied on the random assignment, so both angular and react would get the 2 month expiry.

maxAge should probably be added to the explicit choice functions rather than the base constant.

Reply:
oops sorry, I got confused between this and the admin side. my bad. respondent should be session expiry unless explicitly opting for Angular.

> This will cause the percentage to increase in favour of react more than what the rollout threshold suggests.

yup, I wrote about this in https://github.com/opengovsg/FormSG/issues/4217. I think on the admin side I'd rather have the deviation than have a situation where admins are randomly flipping back and forth between React and Angular. discuss later!

Reply:
Based on our discussion on cookie expiry dates:

**Public respondents**
- random assignment: session cookie
- personal choice - react to angular: 2 months

**Admins**
- random assignment
  - react: 2 months
  - angular: session
- personal choice - react to angular: 2 months
- personal choice - angular to react: 2 months

Reply:
I've removed `maxAge` from the base cookie options, and have added new `ADMIN_COOKIE_OPTIONS_WITH_EXPIRY` and `RESPONDENT_COOKIE_OPTIONS_WITH_EXPIRY` for the personal choice cases, since it's a standard 2 months for both react to angular and vice versa switches.

TODO: Implement random assignment + differentiate random assignment env cookie expiry for admins. 
Currently admins are on Angular by default, and see React if they opt in.
Have left it since it's unrelated to this PR, will open a separate issue